server/index.js

```diff
@@ -223,7 +223,16 @@ export default class Server {
           }
         }
 
-        await renderScript(req, res, page, this.renderOpts)
+        let p = join(this.dir, this.dist, 'bundles', 'pages', paths.join('/'))
+        if (!fs.existsSync(`${p}.js`)) {
+          p = join(p, 'index') // It's possible to have index.js in a subfolder
```

Comment:
This is not possible. We handle this inside the webpack `PagesPlugin`.

Reply:
It's definitely possible. Happens when you do this:

`pages/day/index.js` instead of `pages/day.js`

Reply:
@timneutkens If there's a file called `pages/day/index.js`, it should renamed into `pages/day.js`. We can't have both. We prefer `pages/day.js`.

Earlier we only did this for client side since server setup is different. Now since both are the same, Pages plugin should rename them properly. Otherwise it's an bug in PagesPlugin.

Reply:
@timneutkens here's the logic I mentioned: https://github.com/zeit/next.js/blob/canary/server/build/plugins/pages-plugin.js#L26

Reply:
It seems like a bug with PagesPlugin.
Ref: https://zeit.slack.com/archives/C0DNEN0SH/p1516197323000242
(Internal Discussion)